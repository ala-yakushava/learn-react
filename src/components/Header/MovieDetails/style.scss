@import "../../../styles/theme";

.MovieDetails {
  display: flex;
  padding-top: 40px;
  padding-bottom: 40px;

  &_picture {
    min-width: 350px;
    height: 500px;
    margin-right: 50px;
    background-color: $grey;
  }

  &_inner {
    padding-top: 40px;
  }

  &_header {
    display: flex;
  }

  &_title {
    margin-bottom: 10px;
  }

  &_vote {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 50px;
    height: 50px;
    margin-left: 20px;
    font-size: 24px;
    line-height: 1;
    color: greenyellow;
    border: 1px solid $white;
    border-radius: 50%;
  }

  &_lead {
    margin: 0;
    font-size: 20px;
    line-height: 1.2;
  }

  &_description {
    font-size: 20px;
    line-height: 1.2;
  }

  &_details {
    font-size: 24px;
    line-height: 1.2;
    color: $red;

    span {
      margin-right: 20px;
    }
  }
}
