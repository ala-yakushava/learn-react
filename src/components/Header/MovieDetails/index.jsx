import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';
import Heading from '../../Heading';

const MovieDetails = ({ movie }) => {
  const {
    title, tagline, overview, runtime, vote_average: vote, release_date: date, poster_path: poster,
  } = movie;

  const releaseYear = date ? date.split('-')[0] : null;

  return (
    <article className="MovieDetails" data-testid="movie-details">
      <div className="MovieDetails_picture">
        <img src={poster} alt={`Poster for the movie ${title}`} />
      </div>
      <div className="MovieDetails_inner">
        <header className="MovieDetails_header">
          <Heading className="MovieDetails_title">{ title }</Heading>
          <span className="MovieDetails_vote">{ vote }</span>
        </header>
        <p className="MovieDetails_lead">{ tagline }</p>
        <p className="MovieDetails_details">
          <span>{ releaseYear }</span>
          <span>{ `${runtime} min` }</span>
        </p>
        <p className="MovieDetails_description">{ overview }</p>
      </div>
    </article>
  );
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    runtime: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
  }),
};

MovieDetails.defaultProps = {
  movie: {},
};

export default MovieDetails;
