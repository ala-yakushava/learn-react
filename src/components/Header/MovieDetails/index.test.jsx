import React from 'react';
import renderer from 'react-test-renderer';

import MovieDetails from './index';

describe('Components - MovieDetails', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<MovieDetails />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
