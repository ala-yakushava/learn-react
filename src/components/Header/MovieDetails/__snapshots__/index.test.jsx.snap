// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - MovieDetails should match snapshot 1`] = `
<article
  className="MovieDetails"
  data-testid="movie-details"
>
  <div
    className="MovieDetails_picture"
  >
    <img
      alt="Poster for the movie undefined"
    />
  </div>
  <div
    className="MovieDetails_inner"
  >
    <header
      className="MovieDetails_header"
    >
      <div
        className="Heading MovieDetails_title"
      />
      <span
        className="MovieDetails_vote"
      />
    </header>
    <p
      className="MovieDetails_lead"
    />
    <p
      className="MovieDetails_details"
    >
      <span />
      <span>
        undefined min
      </span>
    </p>
    <p
      className="MovieDetails_description"
    />
  </div>
</article>
`;
